controllers/nnf_workflow_controller_helpers.go

```diff
@@ -1057,106 +1058,50 @@ func (r *NnfWorkflowReconciler) createOrUpdateContainerJobsIfNecessary(ctx conte
-				// with a new IP. A retry is not the same as a restart.  If we set this to
-				// OnFailure, the pods will truly restart but we will lose any log history outside
-				// of (kubectl logs --previous).
-				//
-				// TODO: consider allowing `OnFailure` as well and make it configurable via the
-				// containerprofile. However, it might be a good idea not to allow this for the
-				// reasons above.
-				podSpec.RestartPolicy = corev1.RestartPolicyNever
-
-				// Because the IP changes, we need to have deterministic hostname for the pod
-				podSpec.Hostname = nnfNode
-				podSpec.Subdomain = workflow.Name // service name == workflow name
-
-				// In our case, the target is only 1 node for the job, so a restartPolicy of Never
-				// is ok because any retry (i.e. new pod) will land on the same node.
-				podSpec.NodeSelector = map[string]string{"kubernetes.io/hostname": nnfNode}
-				podSpec.Tolerations = []corev1.Toleration{
-					{
-						Effect:   "NoSchedule",
-						Key:      "cray.nnf.node",
-						Operator: "Equal",
-						Value:    "true",
-					},
-				}
-
-				// TODO Volumes
-				// podSpec.Volumes
-				// ssh is necessary for mpi
-				// setupSSHAuthVolumes(manager, podSpec)
-				// setupLustreVolumes(ctx, manager, podSpec, filesystems.Items)
-				// TODO: add Volumes to DS, append if customer specifies; check for conflicts
-				// vol := corev1.Volume{
-				// 	Name: "foo-local-storage",
-				// 	VolumeSource: corev1.VolumeSource{
-				// 		HostPath: &corev1.HostPathVolumeSource{
-				// 			Path: "/path/to/thing"}}}
-				// podTemplateSpec.Spec.Volumes = append(podTemplateSpec.Spec.Volumes, vol)
-				// TODO: get mounts from workflow
-			}
-
-			// This is a timeout for the job, if set, when the deadline hits, the job will kill all
-			// pods and fail the job.
-			if profile.Data.ActiveDeadlineSeconds > 0 {
-				job.Spec.ActiveDeadlineSeconds = &profile.Data.ActiveDeadlineSeconds
-			}
+		// Get the volumes to mount into the containers
+		volumes, err := r.getContainerVolumes(ctx, workflow, dwArgs, nnfNode)
+		if err != nil {
+			return nnfv1alpha1.NewWorkflowErrorf("could not determine the list of volumes need to create container job %s:", job.Name).WithError(err).WithFatal()
+		}
 
-			// This defaults to 6 and is the maximum number of pod started before considering the
-			// job failed. I don't believe this can be turned off.
-			job.Spec.BackoffLimit = &profile.Data.BackoffLimit
+		// Set the job's podspec from the container profile and add in volumes
+		r.setContainerJobPodSpec(ctx, workflow.Name, job, *profile, nnfNode, volumes)
 
-			return nil
+		// This is a timeout for the job, if set, when the deadline hits, the job will kill all
+		// pods and fail the job.
+		if profile.Data.ActiveDeadlineSeconds > 0 {
+			job.Spec.ActiveDeadlineSeconds = &profile.Data.ActiveDeadlineSeconds
 		}
+		// This defaults to 6 and is the maximum number of pod started before considering the
```

Comment:
In our case, we're using a restart policy of `Never`. Using `Never` actually causes a new pod to start rather than trying to restart one that failed. The reasoning for this is to ensure the pod history/logs are easily found when pods fail.

So this number means that it will retry 6 times. In other words, retry (by starting a new pod) up to 6 times before the job stops firing up new pods and considers the job failed.

Reply:
Maybe you could give a link to documentation here that I could read to get the information you just put in this review comment? :)

Reply:
Here's a comment I left in the container PR to explain this to Nate. It's confusing: https://github.com/NearNodeFlash/NearNodeFlash.github.io/pull/26#discussion_r1089460308. There some links to documentation in there as well.

I'd be happy to chat about this as well. It might be easier to explain over the phone.

Reply:
Maybe put that link into this comment?

Reply:
Will do.